lib/pbench/test/unit/common/test_utils.py

```diff
@@ -18,3 +20,47 @@ def test_md5sum():
         assert (
             hash_md5 == expected_hash_md5
         ), f"Expected MD5 '{expected_hash_md5}', got '{hash_md5}'"
+
+
+class TestCleanup:
+    def test_contruct(self, caplog):
+        logger = logging.getLogger("test_construct")
+        c = Cleanup(logger)
+        assert isinstance(c.actions, collections.deque)
+        assert len(c.actions) == 0
+        assert c.logger is logger
+
+    def test_add(self, caplog):
+        class Test:
+            def delete(self):
+                pass
+
+        logger = logging.getLogger("test_construct")
+        c = Cleanup(logger)
+        t = Test()
+        c.add(t.delete)
+        assert c.actions.pop().action == t.delete
+
+    def test_cleanup(self, caplog):
+        class Test:
+            def __init__(self):
+                self.called = []
+
+            def d1(self):
+                self.called.append("d1")
+
+            def d2(self):
+                self.called.append("d2")
```

Comment:
Perhaps in a new test case rather than here, but it would be good to have an action which raises an exception so that we can prove that subsequent actions are still executed properly.